client/src/Pages/Profile.tsx

```diff
@@ -1,58 +1,91 @@
-import React, { useEffect, useState } from 'react';
-import { WorkCard, UserInfoCard } from '../Components';
+import React, { useEffect, useState, useContext } from 'react';
+import { useParams } from 'react-router-dom';
 import {
-  Works, Request, AllWorks, OnWork, request,
+  UserInfoCard, SpinierComponent, WorkList, Reviews, ErrorComponent,
+} from '../Components';
+import {
+  Works, User, TopTenReviews, ProfileDataProps, WorksProps,
 } from '../utils';
+import { UserContext } from '../Context/LoggedUserContext';
+import { getUserProfileData, getWorksData } from '../Controllers';
+
+const iff = (condition :any, then :any, otherwise :any) => (condition ? then : otherwise);
 
 function Profile() {
-  const [userData, setUserData] = useState({});
-  const [worksData, setWorksData] = useState<AllWorks>({ 1: [] });
+  const [userData, setUserData] = useState<User>();
+  const [worksData, setWorksData] = useState<Works>([]);
+  const [reviewsArray, setReviewsArray] = useState<TopTenReviews[]>([]);
+  const [reviewsAvg, setReviewsAvg] = useState<number>(0);
   const [error, setError] = useState('');
+  const [workError, setWorkError] = useState('');
   const [isLoading, setIsLoading] = useState(true);
-
+  const [workLoading, setWorkLoading] = useState(true);
+  const [page, setPage] = useState(1);
+  const [resultCount, setResultCount] = useState<number>(1);
+  const { id = 1 } = useParams();
+  const userInfo: any = useContext(UserContext);
+  const handlePageChange = (PageNnm: number) => {
+    setPage(PageNnm);
+  };
+  const onSuccess = (
+    user:User,
+    works:Works,
+    reviews:TopTenReviews[],
+    totalReviews:number,
+    count:number,
+  ) => {
+    setReviewsArray(reviews);
+    setReviewsAvg(totalReviews);
+    setUserData(user);
+    setWorksData(works);
+    setResultCount(count);
+  };
+  const onFailed = (err:any) => {
+    setError(err?.data.msg);
+  };
+  const getUserProfileDataParams: ProfileDataProps = {
+    setPage, setIsLoading, successCB: onSuccess, failedCB: onFailed, id: +id,
+  };
+  const onGetWorkSuccess = (works: Works) => {
+    setWorksData(works);
```

Comment:
Let us assume that we are in the first page, Storing thw work data for the first page in the state, And now I want to go to the third page,
SO we're setting the whole state with the new works, And now I want to go to the first page again, So I should send the request again while I had the data before,
This doesn't make any sense
I think the best way to do that is to store the new works beside the one we have related to the page contain,

I think this the best way to store the state
```js
const [works, setWorks] = useState({1: worksForPage1, 3: worksForPage3})
```

And then In the use effect, I should check if I have `works[page]`, then no need to send the request